src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/panel/context.tsx

```diff
@@ -59,6 +74,7 @@ export const PanelProvider: FC<PropsWithChildren<Props>> = ({
   const open = useCallback(
     (navNode: PanelSelectedNode) => {
       setActiveNode(navNode);
+      selectedNodeEl.current = getSelectedNodeEl(navNode);
```

Comment:
I think this looks a lot better, the only so-so place seems to be this  getSelectedNodeEl with direct access to DOM. is there a way we can receive the dom element from the open callback: 

```
const open = useCallback(
     (navNode: PanelSelectedNode, openerEl: HTMLElement) => {
       setActiveNode(navNode);
   selectedNodeEl.current = openerEl;

```

looks like the only place it is called from is src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/navigation_item_open_panel.tsx 


so we need to pass ref from there


wdyt? 


Otherwise lgtm

Reply:
@Dosant I agree the right solution should not access the DOM directly. I took a look and we can use the elements that are passed around from the click handlers. The clicked element isn't always the element that represents the "active node," so I'm using `element.closest(selector)` to find it. See:
* 5e1c67c2d72: using the nav node element closest to the clicked target element to set as the selected node element
* b1117e85a9e: cleanup - I'm not sure how we can set the selected node element from `useEffect`, but I don't think we need it.